prov/efa/src/efa_rma.c

```diff
@@ -37,8 +37,23 @@
 #include <ofi_iov.h>
 #include "efa.h"
 
-static
-ssize_t efa_rma_post_read(struct efa_ep *ep, const struct fi_msg_rma *msg, uint64_t flags)
+
+/*
+ * efa_rma_post_read() will post a read request.
+ *
+ * Input:
+ *     ep: endpoint
+ *     msg: read operation information
+ *     flags: currently no flags is taken
+ *     self_comm: indicate whether the read is toward
+ *                the end point itself. If self_comm is true,
+ *                src_addr should be FI_ADDR_NOTAVAIL
```

Comment:
Is this true for the self address handle? `FI_ADDR_NOTAVAIL` from the device also means we haven't inserted that address yet.

Reply:
Not sure I understand the question? 

The self_ah is an ibv_ah, 

Device does not understand libfabric address. It can only understand ibv_ah. The efa lower provider translate libfabric address to ibv_ah, and pass it to device. We set side an ibv_ah just for self communication, which is not associated with any libfabric address, therefore when caller want to use self comm, the caller must specify FI_ADDR_NOTAVAIL as address.

Reply:
That makes sense to me.

I'm getting hung up by this comment 
```
If self_comm is true,
 *                src_addr should be FI_ADDR_NOTAVAIL
```

We have other parts of the code that have variables named src_addr where the value can be FI_ADDR_NOTAVAIL for other reasons -- specifically when the address is not available.

Reply:
I see. I should have said "msg->addr", not src_addr, will fix the comment.